src/Libraries/CoreNodeModels/DefineData.cs

```diff
@@ -135,13 +172,31 @@ public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode
         /// <param name="data"></param>
         private void DataBridgeCallback(object data)
         {
-            var inputs = data as ArrayList;
+            if (data == null) return;
 
-            var inputObject = inputs[0];
+            (bool IsValid, bool UpdateList, DataNodeDynamoType InputType) resultData = (ValueTuple<bool, bool, DataNodeDynamoType>)data;
 
-            if (!InPorts[0].IsConnected)
+            if (IsAutoMode && resultData.UpdateList)
             {
-                return;
+                IsList = !IsList;
+            }
+
+            if (!resultData.IsValid)
+            {
+                if (IsAutoMode)
+                {
+                    // Assign to the correct value, if the object was of supported type
+                    if (resultData.InputType != null)
+                    {
+                        var index = Items.IndexOf(Items.First(i => i.Name.Equals(resultData.InputType.Name)));
+                        SelectedIndex = index;
+                    }
+                }
+                else
+                {
+                    // Throw an exception/warning and go back to the default dropdown value
+                    SelectedIndex = 0;
```

Comment:
Why do we change the selected item in the list when the input is invalid and we're not in auto mode? Shouldn't the selected item remain the same here?

Reply:
Right on!